### PR TITLE
Acl improvements

### PR DESCRIPTION
**Pull request by Andrews548**
ACL Execution Provider improvements:
- fix gemm corner cases by replacing acl implementation with fully connected layer
- expose acl version to cmake, the switch to build.sh will be made after adding python support
- added the optimization for convolution-activation fusion

### CODE REVIEW

**Comment by jywu-msft** [comment]
Thanks! We'll take a look. @prabhat00155 FYI

File: BUILD.md [hunk 1]
```diff
@@ -403,6 +403,8 @@ alias cmake="/usr/bin/cmake -DCMAKE_TOOLCHAIN_FILE=$OECORE_NATIVE_SYSROOT/usr/sh
 ```
 cmake ../onnxruntime-arm-upstream/cmake -DONNX_CUSTOM_PROTOC_EXECUTABLE=/usr/bin/protoc -Donnxruntime_RUN_ONNX_TESTS=OFF -Donnxruntime_GENERATE_TEST_REPORTS=ON -Donnxruntime_DEV_MODE=ON -DPYTHON_EXECUTABLE=/usr/bin/python3 -Donnxruntime_USE_CUDA=OFF -Donnxruntime_USE_NSYNC=OFF -Donnxruntime_CUDNN_HOME= -Donnxruntime_USE_JEMALLOC=OFF -Donnxruntime_ENABLE_PYTHON=OFF -Donnxruntime_BUILD_CSHARP=OFF -Donnxruntime_BUILD_SHARED_LIB=ON -Donnxruntime_USE_EIGEN_FOR_BLAS=ON -Donnxruntime_USE_OPENBLAS=OFF -Donnxruntime_USE_ACL=ON -Donnxruntime_USE_DNNL=OFF -Donnxruntime_USE_MKLML=OFF -Donnxruntime_USE_OPENMP=ON -Donnxruntime_USE_TVM=OFF -Donnxruntime_USE_LLVM=OFF -Donnxruntime_ENABLE_MICROSOFT_INTERNAL=OFF -Donnxruntime_USE_BRAINSLICE=OFF -Donnxruntime_USE_NUPHAR=OFF -Donnxruntime_USE_EIGEN_THREADPOOL=OFF -Donnxruntime_BUILD_UNIT_TESTS=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo
 ```
+The ```-Donnxruntime_USE_ACL=ON``` option will use, by default, the 19.05 version of the Arm Compute Library. To set the right version you can use:
+```-Donnxruntime_USE_ACL_1902=ON```, ```-Donnxruntime_USE_ACL_1905=ON``` or ```-Donnxruntime_USE_ACL_1908=ON```;
```
Comment (HectorSVC) [inline BUILD.md:407]: We should make sure the ACL version matches with these option right? higher version is not supported? There's ACL 20.02 now.

Comment (mcaraman) [reply BUILD.md:407]: We only support the ACL versions integrated in i.MX SDK toolchain, the latest being 19.08. Work is in progress to add version 20.02 into the new toolchain and to support it with ONNX Runtime.
If the specified option does not match the ACL version we get a compile error. We may automate the version detection by checking the related libarmnn.so.wx.yz file name in the SDK sysroot. ACL also exposes a function API for versioning, but we will need an Arm emulator to run it, which is overkill.

**Pull request merged**